### ISSUE TITLE
[BUG] Intermittantly failing to skip FreeVee ads Prime Video

### ISSUE THREAD

**Issue by mikeloeven** [Dreamlinerm/Netflix-Prime-Auto-Skip#75]
**Describe the bug**
Seems some of the FreeVee midroll ads on prime video are not properly skipped and will play normally. Feature does appear partially functional as in player controls dont get locked out so you can still manually skip it however it should be doing this automatically. 

**To Reproduce**
unfortunately it's intermittent sometimes works some times doesn't I haven't been able to reliably reproduce 

**Expected behavior**
midroll ads skipped without manual fast forwarding

**Background information:**
 - URL: `https://www.amazon.com/gp/video/detail/B0B8THXGZN/ref=atv_mv_hom_c_rVC5eJ_brws_9_1?jic=36%7CCgtmcmVld2l0aGFkcxIMc3Vic2NyaXB0aW9u`
 - OS: Windows 10
 - Browser Chrome
 - Addon-Version 1.0.70

**Comment by Dreamlinerm** [comment]
> **Describe the bug** Seems some of the FreeVee midroll ads on prime video are not properly skipped and will play normally. Feature does appear partially functional as in player controls dont get locked out so you can still manually skip it however it should be doing this automatically.
> 
> **To Reproduce** unfortunately it's intermittent sometimes works some times doesn't I haven't been able to reliably reproduce
> 
> **Expected behavior** midroll ads skipped without manual fast forwarding
> 
> **Screenshots** If applicable, add screenshots
> 
> **Background information:**
> 
>     * URL: `https://www.amazon.com/gp/video/detail/B0B8THXGZN/ref=atv_mv_hom_c_rVC5eJ_brws_9_1?jic=36%7CCgtmcmVld2l0aGFkcxIMc3Vic2NyaXB0aW9u`
> 
>     * OS: Windows 10
> 
>     * Browser Chrome
> 
>     * Addon-Version 1.0.70

Hmm I will look into the code if anything could cause this, but since this bug is intermittend its hard for me to tell what could be the cause.

**Comment by Dreamlinerm** [comment]
Does this midroll ad have a skip button? if the non skipped one has one that may be the reason it wont get skipped.

**Comment by Dreamlinerm** [comment]
I skipped a lot of ads and didnt seem to run into the same issue. 
if you want we can look into your browser on discord when you have some tme: https://discord.gg/NSZVDWum

**Comment by Dreamlinerm** [comment]
What show did the problem occur. I learned that some behaviours  can only be seen in specific shows.

**Comment by swi-da** [comment]
I've seen the described behavior several times in the last days and was now able to reproduce it.
In this video (newly added to Freevee) an Ad starts at 03:16 and does not have a skip button.

https://www.amazon.de/gp/video/detail/B00ERL9BFI/ref=atv_dp_season_select_s7?jic=8%7CEgNhbGw%3D

You can manually skip it in the timeline, but it seems like they just got rid of the skip button for some ads.

In my case its Browser Firefox, Addon version 1.0.74.

**Comment by Dreamlinerm** [comment]
that's great, I am looking into it if I can replicate it 

> I've seen the described behavior several times in the last days and was now able to reproduce it. In this video (newly added to Freevee) an Ad starts at 03:16 and does not have a skip button.
> 
> https://www.amazon.de/gp/video/detail/B00ERL9BFI/ref=atv_dp_season_select_s7?jic=8%7CEgNhbGw%3D
> 
> You can manually skip it in the timeline, but it seems like they just got rid of the skip button for some ads.
> 
> In my case its Browser Firefox, Addon version 1.0.74.

**Comment by Dreamlinerm** [comment]
if you say they got rid of the skip button do you mean the freevee ads or the self ads?

> I've seen the described behavior several times in the last days and was now able to reproduce it. In this video (newly added to Freevee) an Ad starts at 03:16 and does not have a skip button.
> 
> https://www.amazon.de/gp/video/detail/B00ERL9BFI/ref=atv_dp_season_select_s7?jic=8%7CEgNhbGw%3D
> 
> You can manually skip it in the timeline, but it seems like they just got rid of the skip button for some ads.
> 
> In my case its Browser Firefox, Addon version 1.0.74.

**Comment by swi-da** [comment]
The Freevee Ads for Third Parties, for self ads I'm not entirely sure if I've seen it happening with them.
Next segment starts at 13:33, always ,the same timestamp.

**Comment by Dreamlinerm** [comment]
> The Freevee Ads for Third Parties, for self ads I'm not entirely sure if I've seen it happening with them. Next segment starts at 13:33, always ,the same timestamp.

can you come into the discord and show me because I really cant seem to get this issue. I dont know why. https://discord.gg/4w5eT2me

**Comment by Dreamlinerm** [comment]
For some ads there is no Skip in ... seconds text and you can freely skip them for some reason.
If this bug wont get fixed by amazon I maybe will fix it later. This seems more like a Amazon rather than a me problem.
<img width="1199" alt="image" src="https://github.com/Dreamlinerm/Netflix-Prime-Auto-Skip/assets/90410608/c161d99b-5e21-4ff0-9f81-f35b83852684">

**Comment by MurphyDoe0815** [comment]
the same error for me!

here at netflix and adverts are skipped and stop. manually click on continue and everything works again.

but here i don't know exactly what's going on. according to the adverts in the bar it should happen more often in the film. but i only have it once in the film? 



configuration: raspi4 with raspi os 32 bit (because of drm) and default browser. films run great and without problems. add on is up to date. (installed yesterday)

everything that is possible with skip is activated for netflix.



sorry, but i didn't want to post extra ...

**Comment by Dreamlinerm** [comment]
> ev

What do you mean with manually click on continue and what do you mean with skipped and stop? 
So it only shows the first ad even though the scroll bar shows more ads, that's fine I guess there is no problem there right?.

**Comment by Dreamlinerm** [comment]
> the same error for me!
> 
> here at netflix and adverts are skipped and stop. manually click on continue and everything works again.
> 
> but here i don't know exactly what's going on. according to the adverts in the bar it should happen more often in the film. but i only have it once in the film?
> 
> configuration: raspi4 with raspi os 32 bit (because of drm) and default browser. films run great and without problems. add on is up to date. (installed yesterday)
> 
> everything that is possible with skip is activated for netflix.
> 
> sorry, but i didn't want to post extra ...

**Comment by MurphyDoe0815** [comment]
a series has a time bar. there are 4 interruptions displayed in a different colour. i assume this is the commercial break.



but i am only interrupted once and the advert (here for chrome, for example) is fast-forwarded and you see it briefly at increased speed. and then everything stops and waits for an input.



the display is only the end of the advert and you then have to press start or skip with the mouse, otherwise nothing wants to happen.

**Comment by Dreamlinerm** [comment]
> a series has a time bar. there are 4 interruptions displayed in a different colour. i assume this is the commercial break.
> 
> but i am only interrupted once and the advert (here for chrome, for example) is fast-forwarded and you see it briefly at increased speed. and then everything stops and waits for an input.
> 
> the display is only the end of the advert and you then have to press start or skip with the mouse, otherwise nothing wants to happen.

ok I will try to fix this

**Comment by Dreamlinerm** [comment]
> a series has a time bar. there are 4 interruptions displayed in a different colour. i assume this is the commercial break.
> 
> but i am only interrupted once and the advert (here for chrome, for example) is fast-forwarded and you see it briefly at increased speed. and then everything stops and waits for an input.
> 
> the display is only the end of the advert and you then have to press start or skip with the mouse, otherwise nothing wants to happen.

does this happen alot or occasionally

**Comment by Dreamlinerm** [comment]
> a series has a time bar. there are 4 interruptions displayed in a different colour. i assume this is the commercial break.
> 
> but i am only interrupted once and the advert (here for chrome, for example) is fast-forwarded and you see it briefly at increased speed. and then everything stops and waits for an input.
> 
> the display is only the end of the advert and you then have to press start or skip with the mouse, otherwise nothing wants to happen.

I uploaded version 81 can you check if I have successfully fixed the issue?

**Comment by MurphyDoe0815** [comment]
unfortunately not: Version 1081



adverts are skipped, but remain frozen in the last frame. a click on continue is addressed, but does not react. almost as if a layer was placed over it.



so i have to go all the way out and restart the film.

at first i thought it was due to the slower raspi, but that's not the case. 

the faster laptop as well.



by the way, a question: do you know smarttube ?

wouldn't that be an option with the possibility of logging in?

it filters excellently and could be installed anywhere.
![Bildschirmfoto vom 2024-01-21 06-34-12](https://github.com/Dreamlinerm/Netflix-Prime-Auto-Skip/assets/99426089/7a893f13-1529-4a9f-97a2-f8df857fc8d1)

**Comment by Dreamlinerm** [comment]
> unfortunately not: Version 1081
> 
> adverts are skipped, but remain frozen in the last frame. a click on continue is addressed, but does not react. almost as if a layer was placed over it.
> 
> so i have to go all the way out and restart the film.
> 
> at first i thought it was due to the slower raspi, but that's not the case.
> 
> the faster laptop as well.
> 
> by the way, a question: do you know smarttube ?
> 
> wouldn't that be an option with the possibility of logging in?
> 
> it filters excellently and could be installed anywhere. ![Bildschirmfoto vom 2024-01-21 06-34-12](https://private-user-images.githubusercontent.com/99426089/298362742-7a893f13-1529-4a9f-97a2-f8df857fc8d1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU4Mzg0MzEsIm5iZiI6MTcwNTgzODEzMSwicGF0aCI6Ii85OTQyNjA4OS8yOTgzNjI3NDItN2E4OTNmMTMtMTUyOS00YTlmLTk3YTItZjhkZjg1N2ZjOGQxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIxVDExNTUzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJkZWFkYTYxMjNjYjA5Njc3ZDczOTQ4YTIxZDNkNGZjOTIzZWQ2YWFhYmU5NzY0ZWFmZDAxOTIyMDM2ZjRkNjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.aRhZkAzdqgaHgAm810mHNbhw4Bu3l3gPfkh1eG1kkIA)

Can you maybe show me the issue over discord?
https://discord.gg/AB6h2Vus
So basically the page becomes unresponsive after the ad is finished

**Comment by MurphyDoe0815** [comment]
Discord wasn't my friend today and was causing problems.

But never mind - I think I've found it and it's no longer there.



It was probably due to the raspberry pi's sd card.

Everything may be writing far too slowly to the pi.

Now I have booted an external ssd with OS and lo and behold: everything runs much faster and even films and series no longer stall easily.

Freevee also works well.



But I will keep an eye on it. 

Many thanks for your help - great project !!!
